test/meson.build

```diff
@@ -1,3 +1,4 @@
+has_dlvsym = cc.has_function('dlvsym', prefix : '#include<dlfcn.h>')
```

Comment:
I think you need to also define `_GNU_SOURCE`, according to the man page for `dlvsym`:

```
has_dlvsym = cc.has_function(
  'dlvsym',
  prefix: '''#define _GNU_SOURCE
#include <dlfcn.h>''',
)
```